packages/toolkit/src/shared/telemetry/util.ts

```diff
@@ -90,6 +92,40 @@ export async function getUserAgent(
     return pairs.join(' ')
 }
 
+type EnvType =
+    | 'cloud9'
+    | 'cloud9-codecatalyst'
+    | 'amazon-cloud9'
+    | 'cloud9-ec2'
+    | 'codecatalyst'
+    | 'local'
+    | 'ec2'
+    | 'sagemaker'
+    | 'test'
+    | 'other'
+
+export function getComputeEnvType(): EnvType {
+    if (isCloud9('classic')) {
+        return 'cloud9'
+    } else if (isCloud9('codecatalyst')) {
+        return 'cloud9-codecatalyst'
+    } else if (isInDevEnv()) {
+        return 'codecatalyst'
+    } else if (isCn()) {
+        return 'amazon-cloud9'
```

Comment:
what is the difference between `amazon-cloud9` and `cloud9`? I dont see the former defined in the telemetry .json in this PR.

Also, `isCn()` is checking that the C9 or sagemaker instance is in China like Nik mentioned. Do you need this distinguished for telemetry?

Reply:
since they are two different checks in code, distinguished things are helpful during debugging

Reply:
I would suggest removing this check- `amazon-cloud9` isn't descriptive of what this is distinguishing. As far as I understand, an `amazon-cloud9` type would be reported if its sagemaker instance in China only. The if/else statement would swallow up the cloud9 in China cases.

If you want to know to know if the env is hosted in China or not, add a separate telemetry field. You could add new env types such as `cloud9-cn`, `cloud9-codecatalyst-cn`, `sagemaker-cn` as well, but this seems unnecessary.